pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -156,10 +156,15 @@ public abstract class BaseTableDataManager implements TableDataManager {
   protected volatile boolean _shutDown;
   protected volatile boolean _isDeleted;
 
+  // Executor service to process segment refresh if asynchronous handling is enabled
+  @Nullable
+  protected ExecutorService _segmentRefreshExecutor;
```

Comment:
I think all existing tests will have _segmentRefreshExecutor disabled. So we need a test where this is enabled and we validate that segments are eventually refreshed.

Reply:
done, added tests